src/java/org/apache/cassandra/cql3/constraints/ColumnConstraints.java

```diff
@@ -18,24 +18,42 @@
 
 package org.apache.cassandra.cql3.constraints;
 
-
 import org.apache.cassandra.cql3.CqlBuilder;
+import org.apache.cassandra.db.marshal.AbstractType;
+import org.apache.cassandra.db.marshal.CompositeType;
+import org.apache.cassandra.db.marshal.DynamicCompositeType;
+import org.apache.cassandra.db.marshal.MapType;
+import org.apache.cassandra.db.marshal.TupleType;
+import org.apache.cassandra.db.marshal.UserType;
 import org.apache.cassandra.io.IVersionedAsymmetricSerializer;
 import org.apache.cassandra.io.IVersionedSerializer;
 import org.apache.cassandra.io.util.DataInputPlus;
 import org.apache.cassandra.io.util.DataOutputPlus;
 import org.apache.cassandra.schema.ColumnMetadata;
-import org.apache.cassandra.schema.TableMetadata;
+import org.apache.cassandra.transport.DataType;
+import org.apache.cassandra.transport.ProtocolVersion;
 
 import java.io.IOException;
 import java.util.ArrayList;
 import java.util.Collections;
 import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.function.Function;
+
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.ImmutableSet;
 
 // group of constraints for the column
 public class ColumnConstraints implements ColumnConstraint
 {
-    public static Serializer serializer = new Serializer();
+    public static final Serializer serializer = new Serializer();
+
+    private static final Set<Class<? extends AbstractType>> complexTypes = ImmutableSet.of(MapType.class,
```

Comment:
`complex` means differently. `columnMetadata.isComplex() == true` indicate the value type is collection and UDT. 

Let's not override the meaning, which can create confusion for future code readers. 

How about just calling it `UNSUPPORTED_TYPES`? (All in upper-case).